lib/Analysis/TestPasses.cpp

```diff
@@ -250,6 +251,49 @@ void FIRRTLInstanceInfoPass::runOnOperation() {
     printModuleInfo(op, iInfo);
 }
 
+//===----------------------------------------------------------------------===//
+// Op Count
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct TestOpCountAnalysisPass
+    : public PassWrapper<TestOpCountAnalysisPass, OperationPass<ModuleOp>> {
+  MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(TestOpCountAnalysisPass)
+
+  void runOnOperation() override;
+  StringRef getArgument() const override { return "test-op-count-analysis"; }
+  StringRef getDescription() const override {
+    return "Run OpCountAnalysis and show the results.  This pass "
+           "is intended to be used for testing purposes only.";
+  }
+};
+} // namespace
+
+void printOpAndOperandCounts(OpCountAnalysis &opCount) {
+  auto opNames = opCount.getFoundOpNames();
+  // Sort to account for non-deterministic DenseMap ordering
+  llvm::sort(opNames, [](OperationName name1, OperationName name2) {
+    return name1.getStringRef() < name2.getStringRef();
+  });
+  for (auto opName : opNames) {
+    llvm::errs() << opName << ": " << opCount.getOpCount(opName) << "\n";
+    auto operandMap = opCount.getOperandCountMap(opName);
+    // Sort for determinism again
+    llvm::SmallVector<size_t> keys;
+    for (auto pair : operandMap)
+      keys.push_back(pair.first);
+    llvm::sort(keys);
+    for (auto num : keys)
+      llvm::errs() << " with " << num << " operands: " << operandMap[num]
+                   << "\n";
+  }
```

Comment:
I'll leave this format to you, particularly since you mentioned bridging this to JSON or something like that.

I tend to actually export this information directly as YAML (manually, not using the internal library). You may be able to go to JSON directly here if you want.

Regardless, I think the format should be less wordy eventually, e.g.:

```
operations:
  - name: builtin.module
    count: 1
  - name: comb.add
    count: 4
    operandCounts:
      - operands: 1
        count: 2
      - operands: 2
        count: 4
```

Reply:
Yeah, I just put this in for testing so figured something readable in case of debugging failure wouldn't hurt. Happy to trim it down if preferred though.